packages/legend-application/src/components/shared/PackageableElementOptionRenderer.tsx

```diff
@@ -14,23 +14,45 @@
  * limitations under the License.
  */
 
-import type { PackageableElement } from '@finos/legend-graph';
+import type {
+  PackageableElement,
+  GraphManagerState,
+} from '@finos/legend-graph';
 import type { PackageableElementOption } from '../../stores/shared/PackageableElementOption.js';
 
-export const getPackageableElementOptionalFormatter = (props?: {
+export const getPackageableElementOptionFormatter = (props: {
   darkMode?: boolean;
+  graphManagerState: GraphManagerState;
 }): ((
   option: PackageableElementOption<PackageableElement>,
 ) => React.ReactNode) =>
   function PackageableElementOptionLabel(
     option: PackageableElementOption<PackageableElement>,
   ): React.ReactNode {
-    const className = props?.darkMode
+    const className = props.darkMode
       ? 'packageable-element-format-option-label--dark'
       : 'packageable-element-format-option-label';
 
+    const element = option.value;
+
+    const optionColor = props.graphManagerState.isPrimitiveTypeElement(element)
+      ? 'primitive'
```

Comment:
I think `primitive` could be grouped with `system`, and `generated` and `main` should not be grouped together. In fact, when it's from main, we might not even need to have any color band - i.e. we keep the color band of transparent color so we don't make a mess out of the left margin. Also, we don't need any tooltip for main graph elements I think.

Reply:
So the more and more I think about this, I think we should just get the root package here and do a switch case instead. That way we have 2 `switch` blocks: (1) to decide the CSS class, (1) to decide the tooltip message

Reply:
Sounds good! 👍